Student/src/main/java/com/fnp/service/StudentService.java

```diff
@@ -23,47 +28,73 @@ public class StudentService {
 	public Student saveStudent(StudentModel studentModel) {
 
 		Student student = converter.saveConverter(studentModel);
+		
+		Student studentSaved = repository.save(student);
+		redis.opsForHash().put("student", studentSaved.getId(), studentSaved);
+		redis.expire("student",20, TimeUnit.SECONDS);
 
-		Student student1 = repository.save(student);
-		return student1;
+		return studentSaved;
 	}
 
 	public String deleteStudent(int id) {
 
 		if (repository.existsById(id)) {
 			repository.deleteById(id);
-
+			redis.opsForHash().delete("student", id);
 			return "deleted";
 		} else {
 			return "doesNotExist";
 		}
 	}
 
 	public String updateStudent(StudentModel studentModel) {
-
+		
 		if (repository.existsById(studentModel.getId())) {
 			Student student = converter.updateConverter(studentModel);
 			repository.save(student);
-
+			if(redis.hasKey("student")) { 
+				if(redis.opsForHash().hasKey("student", student.getId())){
+					redis.opsForHash().delete("student", student.getId());
+					redis.opsForHash().put("student", student.getId(), student);
+					redis.expire("student",20, TimeUnit.SECONDS);
+				}
+			}else {
+				redis.opsForHash().put("student", student.getId(), student);
+				redis.expire("student",20, TimeUnit.SECONDS);
+
+			}
 			return "Updated Successfully.";
 		} else
 			return "Record does not exist.";
 
 	}
 
-	public Optional<Student> getStudent(int id) {
-
-		if (repository.existsById(id)) {
+	public Student getStudent(int id) {
+		
+		if (redis.opsForHash().hasKey("student", id)) {
+			Student student = (Student) redis.opsForHash().get("student", id);
```

Comment:
type caste required?
is it not returning student object directly?

Reply:
yes it is required.